pkg/controllers/clusterresourceplacement/resource_selector_test.go

```diff
@@ -946,48 +1287,36 @@ func TestSortResource(t *testing.T) {
 		},
-		"should gather selected resources with CRD or Namespace in front  with order, second case": {
-			resources: []*unstructured.Unstructured{crd1, crd2, deployment, namespace2, clusterRole},
-			want:      []*unstructured.Unstructured{namespace2, crd2, crd1, deployment, clusterRole},
+		"should handle multiple known resources, different kinds": {
+			resources: []*unstructured.Unstructured{crd2, crd1, secret2, namespace2, namespace1, secret1},
+			want:      []*unstructured.Unstructured{namespace1, namespace2, secret1, secret2, crd1, crd2},
 		},
-		"should gather selected resources with PersistentVolumeClaim in front with order": {
-			resources: []*unstructured.Unstructured{deployment, pvc, namespace1, role},
-			want:      []*unstructured.Unstructured{namespace1, pvc, deployment, role},
+		"should handle multiple known resources, same kinds with different versions": {
+			resources: []*unstructured.Unstructured{v1beta1Deployment, deployment, limitRange},
+			want:      []*unstructured.Unstructured{limitRange, deployment, v1beta1Deployment},
 		},
-		"should gather selected resources with Secret in front with order": {
-			resources: []*unstructured.Unstructured{deployment, secret, namespace1, crd1, namespace2, role},
-			want:      []*unstructured.Unstructured{namespace2, namespace1, crd1, secret, deployment, role},
+		"should handle multiple unknown resources, same kinds": {
+			resources: []*unstructured.Unstructured{testResource2, testResource1},
+			want:      []*unstructured.Unstructured{testResource1, testResource2},
 		},
-		"should gather selected resources with ConfigMap and Secret in front with order": {
-			resources: []*unstructured.Unstructured{deployment, secret, namespace1, role, configMap, secret2},
-			want:      []*unstructured.Unstructured{namespace1, configMap, secret2, secret, deployment, role},
+		"should handle multiple unknown resources, different kinds": {
+			resources: []*unstructured.Unstructured{testResource1, anotherTestResource},
+			want:      []*unstructured.Unstructured{anotherTestResource, testResource1},
 		},
-		"should gather selected all the resources with the right order": {
-			resources: []*unstructured.Unstructured{configMap, deployment, role, crd1, pvc, secret2, clusterRole, secret, namespace1, namespace2, crd2},
-			want:      []*unstructured.Unstructured{namespace2, namespace1, crd2, crd1, configMap, secret2, secret, pvc, deployment, clusterRole, role},
+		"should handle multiple unknown resources, same kinds with different versions": {
+			resources: []*unstructured.Unstructured{v1beta1AnotherTestResource, anotherTestResource},
+			want:      []*unstructured.Unstructured{anotherTestResource, v1beta1AnotherTestResource},
```

Comment:
Hi Arvind! Just a nit: shall we have a mixed test case added? Apologies if I missed it

Reply:
This test case https://github.com/kubefleet-dev/kubefleet/blob/39dca038703fa2f1d949c9f25fdd391f8c9a2edb/pkg/controllers/clusterresourceplacement/resource_selector_test.go#L1290-L1292 has known, unknown kind objects